File: Schaltjahr.py
year = int(input("Jahr: "))

if year % 4 == 0:
    isLeapYear = True
    
    if year % 100 == 0:
        isLeapYear = False
        
        if year % 400 == 0:
            isLeapYear = True
            
        else:
            isLeapYear = False
        
    else:
        isLeapYear = True

else:
    isLeapYear = False
    
if isLeapYear:
    print(f"{year} ist ein Schaltjahr")
    
else:
    print(f"{year} ist kein Schaltjahr")

File: Schaltjahr.java
import java.util.Scanner;

public class Schaltjahr {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.println("\nJahr:");
        int year = input.nextInt();
        input.nextLine();
        input.close();

        boolean isLeapYear = false;

        // Überprüft ob das eingegebene Jahr durch 4 teilbar ist
        if (year % 4 == 0) {
            isLeapYear = true;

            // Überprüft ob das eingegebene Jahr durch 100 teilbar ist
            if (year % 100 == 0) {
                isLeapYear = false;

                // Überprüft ob das eingegebene Jahr durch 400 teilbar ist
                if (year % 400 == 0) {
                    isLeapYear = true;
                }
            }
        }

        if (isLeapYear) {
            System.out.println(year + " ist ein Schaltjahr");
        } else {
            System.out.println(year + " ist kein Schaltjahr");
        }
    }
}